### PR TITLE
feat: add support to enable FullTimeStamp in logging WIP

### LINKED ISSUE #8523: Ability to enable FullTimeStamp option in logs based in environment variable
# Summary

Be able to enable or disable `FullTimeStamp` in `logrus.Formatter` used for logs.

# Motivation

It can be very useful to have logs that have the `FullTimeStamp` option enabled or not depending on whether an environment variable `ARGOCD_LOG_FORMAT_ENABLE_TIMESTAMP` has the value 1 or not, just as it is possible to indicate the log format through the `ARGOCD_LOG_FORMAT` environment variable between text or json.

# Proposal

It would be necessary to change the following files:
- Add a const to indicate the environment variable name used to enable `FullTimeStamp` option in logs, for that purpose is necessary modify the file `common/common.go` as exposed below:
	
```
// Environment variables for tuning and debugging Argo CD
const (
	// EnvVarSSODebug is an environment variable to enable additional OAuth debugging in the API server
	EnvVarSSODebug = "ARGOCD_SSO_DEBUG"
	// EnvVarRBACDebug is an environment variable to enable additional RBAC debugging in the API server
	EnvVarRBACDebug = "ARGOCD_RBAC_DEBUG"
	// Overrides the location where SSH known hosts for repo access data is stored
	EnvVarSSHDataPath = "ARGOCD_SSH_DATA_PATH"
	// Overrides the location where TLS certificate for repo access data is stored
	EnvVarTLSDataPath = "ARGOCD_TLS_DATA_PATH"
	// Specifies number of git remote operations attempts count
	EnvGitAttemptsCount = "ARGOCD_GIT_ATTEMPTS_COUNT"
	// Specifices max duration of git remote operation retry
	EnvGitRetryMaxDuration = "ARGOCD_GIT_RETRY_MAX_DURATION"
	// Specifies duration of git remote operation retry
	EnvGitRetryDuration = "ARGOCD_GIT_RETRY_DURATION"
	// Specifies fator of git remote operation retry
	EnvGitRetryFactor = "ARGOCD_GIT_RETRY_FACTOR"
	// Overrides git submodule support, true by default
	EnvGitSubmoduleEnabled = "ARGOCD_GIT_MODULES_ENABLED"
	// EnvGnuPGHome is the path to ArgoCD's GnuPG keyring for signature verification
	EnvGnuPGHome = "ARGOCD_GNUPGHOME"
	// EnvWatchAPIBufferSize is the buffer size used to transfer K8S watch events to watch API consumer
	EnvWatchAPIBufferSize = "ARGOCD_WATCH_API_BUFFER_SIZE"
	// EnvPauseGenerationAfterFailedAttempts will pause manifest generation after the specified number of failed generation attempts
	EnvPauseGenerationAfterFailedAttempts = "ARGOCD_PAUSE_GEN_AFTER_FAILED_ATTEMPTS"
	// EnvPauseGenerationMinutes pauses manifest generation for the specified number of minutes, after sufficient manifest generation failures
	EnvPauseGenerationMinutes = "ARGOCD_PAUSE_GEN_MINUTES"
	// EnvPauseGenerationRequests pauses manifest generation for the specified number of requests, after sufficient manifest generation failures
	EnvPauseGenerationRequests = "ARGOCD_PAUSE_GEN_REQUESTS"
	// EnvControllerReplicas is the number of controller replicas
	EnvControllerReplicas = "ARGOCD_CONTROLLER_REPLICAS"
	// EnvControllerShard is the shard number that should be handled by controller
	EnvControllerShard = "ARGOCD_CONTROLLER_SHARD"
	// EnvEnableGRPCTimeHistogramEnv enables gRPC metrics collection
	EnvEnableGRPCTimeHistogramEnv = "ARGOCD_ENABLE_GRPC_TIME_HISTOGRAM"
	// EnvGithubAppCredsExpirationDuration controls the caching of Github app credentials. This value is in minutes (default: 60)
	EnvGithubAppCredsExpirationDuration = "ARGOCD_GITHUB_APP_CREDS_EXPIRATION_DURATION"
	// EnvHelmIndexCacheDuration controls how the helm repository index file is cached for (default: 0)
	EnvHelmIndexCacheDuration = "ARGOCD_HELM_INDEX_CACHE_DURATION"
	// EnvRepoServerConfigPath allows to override the configuration path for repo server
	EnvAppConfigPath = "ARGOCD_APP_CONF_PATH"
	// EnvLogFormat log format that is defined by `--logformat` option
	EnvLogFormat = "ARGOCD_LOG_FORMAT"
	// EnvLogFormatEnableTimestamp to indicate whether or not to enable the FullTimestamp option in log logs
	EnvLogFormatEnableTimestamp = "ARGOCD_LOG_FORMAT_ENABLE_TIMESTAMP"
	// EnvLogLevel log level that is defined by `--loglevel` option
	EnvLogLevel = "ARGOCD_LOG_LEVEL"
	// EnvMaxCookieNumber max number of chunks a cookie can be broken into
	EnvMaxCookieNumber = "ARGOCD_MAX_COOKIE_NUMBER"
	// EnvPluginSockFilePath allows to override the pluginSockFilePath for repo server and cmp server
	EnvPluginSockFilePath = "ARGOCD_PLUGINSOCKFILEPATH"
)
```

- Add in the file `util/log/logrus.go` the necessary changes to enable or disable the `FullTimeStamp` option based in the value of environment variable `ARGOCD_LOG_FORMAT_ENABLE_TIMESTAMP` through a local function, also it could be interesting adding some an extra local function to check the value of the variable `FORCE_LOG_COLORS`:

```
package log

import (
	"fmt"
	"os"
	"strings"

	adapter "github.com/bombsimon/logrusr"
	"github.com/go-logr/logr"
	"github.com/sirupsen/logrus"

	"github.com/argoproj/argo-cd/v2/common"
)

const (
	JsonFormat = "json"
	TextFormat = "text"
)

func NewLogrusLogger(fieldLogger logrus.FieldLogger) logr.Logger {
	return adapter.New(fieldLogger, adapter.WithFormatter(func(val interface{}) string {
		return fmt.Sprintf("%v", val)
	}))
}

// NewWithCurrentConfig create logrus logger by using current configuration
func NewWithCurrentConfig() *logrus.Logger {
	l := logrus.New()
	l.SetFormatter(CreateFormatter(os.Getenv(common.EnvLogFormat)))
	l.SetLevel(createLogLevel())
	return l
}

// CreateFormatter create logrus formatter by string
func CreateFormatter(logFormat string) logrus.Formatter {
	var formatType logrus.Formatter
	switch strings.ToLower(logFormat) {
	case JsonFormat:
		formatType = &logrus.JSONFormatter{}
	case TextFormat:
		formatType = &logrus.TextFormatter{
			ForceColors:   checkForceLogColors(),
			FullTimestamp: checkEnableFullTimestamp(),
		}
	default:
		formatType = &logrus.TextFormatter{
			FullTimestamp: checkEnableFullTimestamp(),
		}
	}

	return formatType
}

func checkForceLogColors() bool {
	return strings.ToLower(os.Getenv("FORCE_LOG_COLORS")) == "1"
}

func checkEnableFullTimestamp() bool {
	return strings.ToLower(os.Getenv(common.EnvLogFormatEnableTimestamp)) == "1"
}

func createLogLevel() logrus.Level {
	level, err := logrus.ParseLevel(os.Getenv(common.EnvLogLevel))
	if err != nil {
		level = logrus.InfoLevel
	}
	return level
}

### PR DESCRIPTION
**Pull request by xoanmm**
Signed-off-by: Xoán Mallón <xoanmallon@gmail.com>

Checklist:

* [x] I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/8523) and discussed it with the community
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 

Closes #8523

### CODE REVIEW

File: common/common.go [hunk 1]
```diff
@@ -194,6 +194,8 @@ const (
 	EnvAppConfigPath = "ARGOCD_APP_CONF_PATH"
 	// EnvLogFormat log format that is defined by `--logformat` option
 	EnvLogFormat = "ARGOCD_LOG_FORMAT"
+	// EnvLogFormatEnableTimestamp to indicate whether or not to enable the FullTimestamp option in log logs
+	EnvLogFormatEnableTimestamp = "ARGOCD_LOG_FORMAT_ENABLE_TIMESTAMP"
```
Comment (chetan-rns) [inline common/common.go:198]: Should it be `FULL_TIMESTAMP` instead of just `TIMESTAMP`?

**Review by chetan-rns — COMMENTED** [review]
I think this variable is applicable for the notification controller as well. But looks like it doesn't use the `CreateFormatter()` and can't enable this option

https://github.com/argoproj/argo-cd/blob/f9b7384bf4d5b0197e726ca19e40518c97fd05fd/cmd/argocd-notification/commands/controller.go#L91

**Comment by SavelyKalinov** [comment]
@chetan-rns As I understand it, the pull request starter is missing. Can I proceed with making the necessary modifications to the pull request?

**Comment by crenshaw-dev** [comment]
@SavelyKalinov please do! You can create a new branch based off @xoanmm's work and open a new PR. GitHub will give them co-author credit automatically.

**Comment by pasha-codefresh** [comment]
@CodiumAI-Agent /review

**Pull request closed without merging**